### DISCUSSION TITLE
Language Server written on C# for VS Code Web Extension

### DISCUSSION THREAD

**Discussion by oyatsuk** [microsoft/vscode-discussions#787] [Extension Development QnA] [answered]
I have been working on language server using C# and than integrating it with VS Code extension. I managed to make everything work for VS Code desktop which uses node js but I am struggling to work with server in VS Code web. According to [example](https://github.com/microsoft/vscode-extension-samples/tree/main/lsp-web-extension-sample) I should use Web Worker to run language server on background. In example server is implemented in js while mine is not.

Is it possible to run Language Server written on C# for VS Code web extension?

**Comment by r3m0t** [comment, marked as answer]
You would need to compile the C# to WASM to run it in the user's web browser. Here's an example, but they didn't use C# so I don't know how similar it would be. https://www.hiro.so/blog/write-clarity-smart-contracts-with-zero-installations-how-we-built-an-in-browser-language-server-using-wasm

If you know browser support is important, it makes more sense to write your LSP in TS from the start.

**Reply by oyatsuk** [reply]
Thanks for clarification
